pkg/networkpath/traceroute/runner.go

```diff
@@ -31,16 +39,46 @@ const (
 	DefaultDelay        = 50 //msec
 	DefaultReadTimeout  = 3 * time.Second
 	DefaultOutputFormat = "json"
+
+	tracerouteRunnerModuleName = "traceroute_runner__"
 )
 
+// Telemetry
+var tracerouteRunnerTelemetry = struct {
+	runs       *telemetry.StatCounterWrapper
+	failedRuns *telemetry.StatCounterWrapper
+}{
+	telemetry.NewStatCounterWrapper(tracerouteRunnerModuleName, "runs", []string{}, "Counter measuring the number of traceroutes run"),
+	telemetry.NewStatCounterWrapper(tracerouteRunnerModuleName, "failed_runs", []string{}, "Counter measuring the number of traceroute run failures"),
+}
+
+// NewRunner initializes a new traceroute runner
+func NewRunner() (*Runner, error) {
+	// TODO: we should probably cache this info
+	// double check the underlying call doesn't
+	// already do this and only check if we're
+	// running on AWS
+	networkID, err := ec2.GetNetworkID(context.Background())
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Is there specific reasons we don't use this higher level abstraction: https://github.com/DataDog/datadog-agent/blob/63a3eca629136e2a11848ed28eb61b1ef59581bd/pkg/util/cloudproviders/network.go#L23-L29

It also works for both EC2/GCE and seems to have caching embedded.